src/components/Avatar/Avatar.jsx

```diff
@@ -40,25 +27,39 @@ export default class Avatar extends React.Component {
   }
 
   render () {
-    const { size, image, upload, onChange } = this.props
+    const { size, image, upload, onChange, username } = this.props
     const style = { backgroundImage: 'url(' + image + ')' }
 
-    if (upload) {
-      return (
-        <div className='upload'>
-          <input name='avatar' type='file' ref='fileDialog'
-            style={{ display: 'none' }} onChange={onChange}
-            accept='.jpg, .jpeg, .png'
-          />
+    return (
+      <div
+        style={{ width: size, height: size }}
+        onClick={upload ? this.openDialog : null}
+        className={classNames(
+          styles.Avatar,
+          { [styles.upload]: upload && (image || username) },
+          { [styles.new]: !image && !username },
```

Comment:
Isn't there supposed to always have a username that is not undefined ?

Reply:
there is the case where we have not entered any username yet, when signing up, that's the only case I think

Reply:
okay ! I didn't think of that case !